pkg/cli/clivalidation.go

```diff
@@ -101,10 +101,34 @@ func RequireEnvironmentName(cmd *cobra.Command, args []string, workspace workspa
 	return environmentName, err
 }
 
+// RequireEnvironmentNameOrID checks if an environment name or ID is provided as a flag or as a default environment in the workspace,
+// and returns an error if neither is present. It also handles any errors that occur while parsing the resource.
+func RequireEnvironmentNameOrID(cmd *cobra.Command, args []string, workspace workspaces.Workspace) (string, error) {
+	environmentNameOrID, err := cmd.Flags().GetString("environment")
+	if err != nil {
+		return "", err
+	}
+
+	// We store the environment id in config
+	if environmentNameOrID == "" && workspace.Environment != "" {
+		environmentNameOrID = workspace.Environment
+	}
+
+	if environmentNameOrID == "" && workspace.IsEditableWorkspace() {
+		// Setting a default environment only applies to editable workspaces
+		return "", fmt.Errorf("no environment name or ID provided and no default environment set, " +
+			"either pass in an environment name or ID or set a default environment by using `rad env switch`")
+	} else if environmentNameOrID == "" {
+		return "", fmt.Errorf("no environment name or ID provided, pass in an environment name or ID")
+	}
```

Comment:
Does this mean that the workspace doesn't also have an environment? Should we add that to the error message?

Reply:
The condition where workspace is editable and not set with env is checked at https://github.com/nithyatsu/radius/blob/1577a407a7079c53d56090882de2b74ceb2e99b8/pkg/cli/clivalidation.go#L120

This is because some of the cli code has the ability to make a fall back workspace to get to the server side  (this just injects kubectx = default kubernetes context) and continue to work if config.yaml is not found. If this is the workspace we are dealing with, it is not editable. Then we cannot set env.  L120 is the more common case.